File: AlhambraXR/Assets/Scripts/Network/SeqJSONMessage.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using UnityEngine;

public class SeqJSONMessage
{
    public const string ACTION_NAME = "SeqMessage";

    //action sequence, it will receive message by sequence
    [SerializeField]
    public int actionSeq;

    //package nums
    [SerializeField]
    public int packNum;

    //notify begin or end
    [SerializeField]
    public int flag;
    public enum ActionTag
    {
        BEGIN=0,
        PROCESSING=1,
        END=2
    }

    public string data;

    public SeqJSONMessage(int actionSeq, int packNum, ActionTag flag, string data)
    {
        this.actionSeq = actionSeq;
        this.packNum = packNum;
        this.flag = (int)flag;
        this.data = data;
    }

    public static List<string> GenerateSeqJSON(
        PackedJSONMessages msgs, int packNum)
    {
        List<string> results = new List<string>();
        for (int i = 0; i < msgs.actions.Count; i++)
        {
            ActionTag flag = ActionTag.PROCESSING;
            if (i == msgs.actions.Count - 1)
            {
                flag = ActionTag.END;
            }else if (i==0)
            {
                flag = ActionTag.BEGIN;
            }
            SeqJSONMessage s = new SeqJSONMessage(i, packNum, flag, msgs.actions[i]);
            results.Add(JSONMessage.ActionJSON(ACTION_NAME, s));
        }
        return results;

    }
}

File: Alhambra_Android/app/src/main/java/com/alhambra/network/SeqJSONMessage.java
package com.alhambra.network;

import java.util.Comparator;

public class SeqJSONMessage implements Comparator<SeqJSONMessage>,Comparable<SeqJSONMessage> {
    public static final String ACTION_NAME = "SeqMessage";

    //action sequence, it will receive message by sequence
    public int actionSeq;

    //package nums
    public int packNum;

    //notify begin or end
    public int flag;

    public String data;
    public static final int BEGIN=0;
    public static final int PROCESSING=1;
    public static final int END=2;

    @Override
    public int compare(SeqJSONMessage o1, SeqJSONMessage o2) {
        return Integer.compare(o1.actionSeq, o2.actionSeq);
    }

    @Override
    public int compareTo(SeqJSONMessage o) {
        return Integer.compare(actionSeq,o.actionSeq);
    }
}
